### PR TITLE
Adopt special-use domain names for documentation purpose

### PR DESCRIPTION
**Pull request by luixxiul**
Closes https://github.com/spantaleev/matrix-docker-ansible-deploy/issues/3625

This PR intends to replace strings which indicate homeserver's domain such as `YOUR_DOMAIN`, `domain.com`, `mydomain.com`, `domain.tld`, and `server.com`, with special-use domain names such as example.com, example.net, and example.org, replacing hard-coded values with the template `{{ matrix_domain }}`.

Hopefully adopting them should make it easier to set up the instance as you just would need to replace all `example.com` strings with your own domain name once for all.

If the placeholder clearly indicates the homeserver, it is replaced with example.com. If it is clear that the placeholder indicates another server, then it is replaced with either example.net or example.org.

To be fair I personally like the expression `DOMAIN` as it stands out, but sticking to the technical standard should be preferred.

### CODE REVIEW

File: docs/configuring-playbook-conduit.md [hunk 1]
```diff
@@ -36,7 +36,7 @@ You will have to manually register appservices using the the [register-appservic
 Find the `registration.yaml` in the `/matrix` directory, for example `/matrix/mautrix-signal/bridge/registration.yaml`, then pass the content to Conduit:
 
 
-    @conduit:your.server.name: register-appservice
+    @conduit:example.com: register-appservice
```
Comment (luixxiul) [inline docs/configuring-playbook-conduit.md:39]: The change should be fine; `@conduit:matrix.example.com` should be incorrect.

Comment (spantaleev) [reply docs/configuring-playbook-conduit.md:39]: Sounds correct

File: docs/configuring-well-known.md [hunk 2]
```diff
@@ -50,15 +50,15 @@ matrix_static_files_file_matrix_support_enabled: true
...
   - matrix_id: "@admin2:{{ matrix_domain }}"
-    email_address: admin2@domain.tld
+    email_address: admin2@example.com
     role: m.role.admin
-  - email_address: security@domain.tld
+  - email_address: security@example.com
     role: m.role.security
 
-matrix_static_files_file_matrix_support_property_m_support_page: "https://example.domain.tld/support"
+matrix_static_files_file_matrix_support_property_m_support_page: "https://matrix.example.com/support"
```
Comment (luixxiul) [inline docs/configuring-well-known.md:61]: This change should also be fine, though I am not perfectly sure.

Comment (spantaleev) [reply docs/configuring-well-known.md:61]: This property points to the "support page".

The example URL points to the base domain (or some other domain) which hosts your organization's support page (if any).

More details can be found on this MSC page: https://github.com/matrix-org/matrix-spec-proposals/blob/hs/proposal-admin-contact-1/proposals/1929-admin-contact.md

So I'd say `example.com/support` is a better address, to clearly indicate that it's not the Matrix homeserver that is hosting this page. You need to host it yourself somewhere.

File: roles/custom/matrix-conduit/templates/conduit.toml.j2 [hunk 3]
```diff
@@ -10,7 +10,7 @@
 # and room IDs. Examples: matrix.org, conduit.rs
 
 # The Conduit server needs all /_matrix/ requests to be reachable at
-# https://your.server.name/ on port 443 (client-server) and 8448 (federation).
+# https://example.com/ on port 443 (client-server) and 8448 (federation).
```
Comment (luixxiul) [inline roles/custom/matrix-conduit/templates/conduit.toml.j2:13]: The change should also be fine, I think.

**Comment by luixxiul** [comment]
I checked the whole change repeatedly and it should be fine now, while it is possible that other strings which have been used to describe domain names are left unchanged :crossed_fingers:

**Comment by spantaleev** [comment]
Thank you for your diligence! :bow:

**Pull request merged**